src/extent.c

```diff
@@ -1930,6 +1980,11 @@ extent_merge_impl(tsdn_t *tsdn, arena_t *arena,
 		return true;
 	}
 
+	if (extent_dumpable_get(a) || extent_dumpable_get(b)) {
+		extent_ensure_dumpable(a);
+		extent_ensure_dumpable(b);
```

Comment:
I'm probably just missing something, but shouldn't we just not merge if the dumbable states differ?